tests/phased_training_test.py

```diff
@@ -0,0 +1,143 @@
+# pylint: disable=W0621 redefined-outer-name
+# First party
+# Standard
+import pathlib
+import random
+import typing
+
+# Third Party
+import click
+
+# Third party
+import pytest
+
+# First Party
+import instructlab.model.train as model_train
+
+FakeEvalFunc = typing.Callable[..., float]
+
+
+@pytest.fixture
+def dir_of_checkpoints(tmp_path: pathlib.Path) -> pathlib.Path:
+    for dir_name in ["fakedir1", "fakedir2"]:
+        (tmp_path / dir_name).mkdir()
+
+    return tmp_path
+
+
+@pytest.fixture
+def dir_of_checkpoints_and_a_file(
+    dir_of_checkpoints: pathlib.Path,
+) -> pathlib.Path:
+    (dir_of_checkpoints / "file.txt").write_text("something")
+    return dir_of_checkpoints
+
+
+@pytest.fixture
+def fake_eval_func() -> FakeEvalFunc:
+    return lambda model: random.randrange(10) * 1.0
```

Comment:
In theory you could mock the actual eval command, but I think this is fine for now!